### DISCUSSION TITLE
The difference between server and local database search

### DISCUSSION THREAD

**Discussion by Wxl980509** [vdemichev/DiaNN#1294] [Q&A]
I used the same version, the same file and setting parameters, and the server version has 2000 more proteins than the local (25%), is this reasonable.
The logs are as follows
IA-NN 1.9.1 (Data-Independent Acquisition by Neural Networks)
Compiled on Jul 15 2024 09:42:01
Current date and time: Tue Nov 19 10:07:53 2024
Logical CPU cores: 112
/diann1.9/diann-1.9.1/diann-linux --dir rawout --fasta ./human_uniprotkb_proteome_UP000005640_2024_08_02.fasta --threads 50.0 --out diannout --fasta-search --qvalue 0.01 --min-pep-len 7 --max-pep-len 30 --min-pr-mz 380 --max-pr-mz 980 --min-pr-charge 1 --max-pr-charge 6 --cut K*,R*,P --missed-cleavages 2 --unimod4 --var-mods 1 --var-mod UniMod:35,15.994915,M --var-mod UniMod:1,42.010565,*n --peptidoforms --reanalyse --relaxed-prot-inf --smart-profiling --gen-spec-lib --reannotate --xic --cont-quant-exclude cRAP --met-excision 

Thread number set to 50
DIA-NN will carry out FASTA digest for in silico lib generation
Output will be filtered at 0.01 FDR
Min peptide length set to 7
Max peptide length set to 30
Min precursor m/z set to 380
Max precursor m/z set to 980
Min precursor charge set to 1
Max precursor charge set to 6
In silico digest will involve cuts at K*,R*
Maximum number of missed cleavages set to 2
Cysteine carbamidomethylation enabled as a fixed modification
Maximum number of variable modifications set to 1
Modification UniMod:35 with mass delta 15.9949 at M will be considered as variable
Modification UniMod:1 with mass delta 42.0106 at *n will be considered as variable
Peptidoform scoring enabled
A spectral library will be created from the DIA runs and used to reanalyse them; .quant files will only be saved to disk during the first step
Heuristic protein grouping will be used, to reduce the number of protein groups obtained; this mode is recommended for benchmarking protein ID numbers, GO/pathway and system-scale analyses
When generating a spectral library, in silico predicted spectra will be retained if deemed more reliable than experimental ones
A spectral library will be generated
Library precursors will be reannotated using the FASTA database
XICs within 10 seconds from the apex will be extracted for each precursor and saved in .parquet format, a folder will be created next to the main report for the XICs storage
Peptides corresponding to protein sequence IDs tagged with cRAP will be excluded from normalisation as well as quantification of protein groups that do not include proteins bearing the tag
N-terminal methionine excision enabled
DIA-NN will optimise the mass accuracy automatically using the first run in the experiment. This is useful primarily for quick initial analyses, when it is not yet known which mass accuracy setting works best for a particular acquisition scheme.
WARNING: FASTA digest mode enabled and raw data are provided, turning on deep learning spectra/RT/IM prediction
WARNING: it is strongly recommended to first generate an in silico-predicted library in a separate pipeline step and then use it to process the raw data, now without activating FASTA digest
The following variable modifications will be scored: UniMod:35 UniMod:1 

DIA-NN 1.9 (Data-Independent Acquisition by Neural Networks)
Compiled on Jun  8 2024 20:00:31
Current date and time: Wed Oct  9 19:07:52 2024
CPU: GenuineIntel Intel(R) Core(TM) i9-10900X CPU @ 3.70GHz
SIMD instructions: AVX AVX2 AVX512CD AVX512F FMA SSE4.1 SSE4.2 
Logical CPU cores: 20
diann.exe --f G:\PRODUCTION\20240929_SNML042824090201_CX_DIAStd\cx240913001.raw  --f G:\PRODUCTION\20240929_SNML042824090201_CX_DIAStd\cx240913002.raw  --f G:\PRODUCTION\20240929_SNML042824090201_CX_DIAStd\cx240913003.raw  --f G:\PRODUCTION\20240929_SNML042824090201_CX_DIAStd\cx240913004.raw  --f G:\PRODUCTION\20240929_SNML042824090201_CX_DIAStd\cx240913005.raw  --f G:\PRODUCTION\20240929_SNML042824090201_CX_DIAStd\cx240913006.raw  --lib D:\protein\production\library_database\human_report-lib.predicted.speclib --threads 3 --verbose 1 --out G:\PRODUCTION\20240929_SNML042824090201_CX_DIAStd\report.tsv --qvalue 0.01 --matrices --out-lib G:\PRODUCTION\20240929_SNML042824090201_CX_DIAStd\report-lib.tsv --gen-spec-lib --reannotate --xic --fasta camprotR_240512_cRAP_20190401_full_tags.fasta --cont-quant-exclude cRAP- --fasta D:\protein\database\common\animal\human_uniprotkb_proteome_UP000005640_2024_08_02.fasta --met-excision --min-pep-len 7 --max-pep-len 30 --min-pr-mz 380 --max-pr-mz 980 --min-pr-charge 1 --max-pr-charge 6 --cut K*,R*,!*P --missed-cleavages 2 --unimod4 --var-mods 1 --var-mod UniMod:35,15.994915,M --var-mod UniMod:1,42.010565,*n --peptidoforms --reanalyse --relaxed-prot-inf --smart-profiling 

Thread number set to 3
Output will be filtered at 0.01 FDR
Precursor/protein x samples expression level matrices will be saved along with the main report
A spectral library will be generated
Library precursors will be reannotated using the FASTA database
XICs within 10 seconds from the apex will be extracted for each precursor and saved in .parquet format, a folder will be created next to the main report for the XICs storage
Peptides corresponding to protein sequence IDs tagged with cRAP- will be excluded from normalisation as well as quantification of protein groups that do not include proteins bearing the tag
N-terminal methionine excision enabled
Min peptide length set to 7
Max peptide length set to 30
Min precursor m/z set to 380
Max precursor m/z set to 980
Min precursor charge set to 1
Max precursor charge set to 6
In silico digest will involve cuts at K*,R*
But excluding cuts at *P
Maximum number of missed cleavages set to 2
Cysteine carbamidomethylation enabled as a fixed modification
Maximum number of variable modifications set to 1
Modification UniMod:35 with mass delta 15.9949 at M will be considered as variable
Modification UniMod:1 with mass delta 42.0106 at *n will be considered as variable
Peptidoform scoring enabled
A spectral library will be created from the DIA runs and used to reanalyse them; .quant files will only be saved to disk during the first step
Heuristic protein grouping will be used, to reduce the number of protein groups obtained; this mode is recommended for benchmarking protein ID numbers, GO/pathway and system-scale analyses
When generating a spectral library, in silico predicted spectra will be retained if deemed more reliable than experimental ones
DIA-NN will optimise the mass accuracy automatically using the first run in the experiment. This is useful primarily for quick initial analyses, when it is not yet known which mass accuracy setting works best for a particular acquisition scheme.
The following variable modifications will be scored: UniMod:35 UniMod:1 
[report.log.txt](https://github.com/user-attachments/files/18003372/report.log.txt)
[diannout.log.txt](https://github.com/user-attachments/files/18003373/diannout.log.txt)

**Comment by vdemichev** [comment]
Hi,

Please address the warnings.

Best,
Vadim

**Reply by Wxl980509** [reply]
Hi,
Thanks for your answer, please which command controls the FASTA digest mode in the server version , and whether I close it, plus the pre-built spectrum library to ensure that the server and the local are basically the same.

**Reply by vdemichev** [reply]
> please which command controls the FASTA digest mode in the server version

Please copy the command formed by the GUI.

**Reply by Wxl980509** [reply]
Hello, 
  I later used the same parameters and libraries for both Liunx and GUI searching. There is not much difference in the amount of identification, but there is some deviation in the quantitative value. However, under the same processing method, more than half or more of the differentially expressed proteins in linux and GUI are different, is this reasonable and can it be applied?
![image](https://github.com/user-attachments/assets/3e37cea8-edcc-48cd-9025-a788a635e7d0)
